packages/lit-dev-content/samples/articles/lit-cheat-sheet/bind-tag-name/input-or-textfield.ts

```diff
@@ -0,0 +1,35 @@
+import { html, LitElement } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { html as staticHTML, StaticValue } from 'lit/static-html.js';
+import { live } from 'lit/directives/live.js';
+
+@customElement('input-or-textfield')
+export class MyElement extends LitElement {
+  // attribute is false because this is a value that can't be serialized to an
+  // HTML attribute
+  @property({ attribute: false }) tagLiteral: StaticValue|null = null;
+  @property() value = '';
+
+  render() {
+    return html`
+      ${
+        // NOTE: the live() directive prevents setting the .value property if
+        // the live value of the input / textfield already matches this.value.
+        // This is important since static html templates should not be thrashed
+        // due to performance concerns.
+        staticHTML`
+          <${this.tagLiteral}
```

Comment:
Why doesn't this use `unsafeStatic` or `literal`?

Reply:
This uses literal in the parent element. Also unsafeStatic is handled in the next example